### DISCUSSION TITLE
Errors in setting up the repository

### DISCUSSION THREAD

**Discussion by deependujha** [timeplus-io/proton#839] [Q&A]
To setup the repository I'm following the [BUILD.md](https://github.com/timeplus-io/proton/blob/develop/BUILD.md) file. First of all it takes a lot of time (primarily bcoz my Linux has 12 GB ram, and 8 cpus, and might not be enough).

And when I run `ninja` command, after 2 hours or so, it fails with `switch missing default labels` or `deprecated warning` or something else. Even running docker fails.

```
 File "/home/deependu/proton/./docker/packager/packager", line 345, in <module>
    run_docker_image_with_env(image_name, args.output_dir, env_prepared, ch_root, args.ccache_dir, args.docker_image_version)
  File "/home/deependu/proton/./docker/packager/packager", line 63, in run_docker_image_with_env
    subprocess.check_call(cmd, shell=True)
  File "/home/deependu/miniconda3/lib/python3.12/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
```

Is there something that I can do?

**Comment by chenziliang** [comment]
Probably related to memory limits hit. According to our experience, you will need 32GB  (16GB at least) for final linking the binary. Maybe it failed in the last mile

**Reply by deependujha** [reply]
thanks for the quick reply. I'll try setting it up in google idx or some cloud ide. Let's see if that works!

**Reply by chenziliang** [reply]
Please keep us posted, thanks @deependujha
